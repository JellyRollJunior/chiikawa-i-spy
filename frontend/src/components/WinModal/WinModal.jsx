import { useState } from 'react';
import { useNavigate } from 'react-router';
import { IconWrapper } from '../IconWrapper/IconWrapper';
import { makeRequest } from '../../utils/requests';
import { getUrl } from '../../utils/serverUrl';
import styles from './WinModal.module.css';
import shared from '../../styles/shared.module.css';

const WinModal = ({ time }) => {
  const navigate = useNavigate();
  const [name, setName] = useState('Anonymous Mob Creature');
  const [error, setError] = useState(null);

  const handleSubmit = async (event) => {
    event.preventDefault();
    try {
      const token = localStorage.getItem('token');
      await makeRequest(getUrl('/winners'), {
        mode: 'cors',
        method: 'PUT',
        headers: {
          Authorization: `Bearer ${token}`,
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          name,
        }),
      });
      navigate('/');
    } catch (error) {
      console.error(error);
      setError('Unable to set name. Please try again.');
    }
  };

  return (
    <div className={styles.modalWrapper}>
      <div className={`${shared.card} ${styles.modal} ${shared.fadeIn}`}>
        {error && <h3 className={shared.error}>{error}</h3>}
        <form className={styles.form} onSubmit={handleSubmit}>
          <IconWrapper size={32}>
            <h2>Winner</h2>
          </IconWrapper>
          <p>
            Wow! You found all the hidden critters in{' '}
            <strong className={shared.bold}>{time} seconds</strong>
          </p>
          <p>Add your username to the leaderboards</p>
          <label className={`${shared.bold} ${shared.marginTopSmall}`} htmlFor="name">Name</label>
          <input
            type="text"
            name="name"
            id="name"
            value={name}
            onChange={(event) => setName(event.target.value)}
            minLength={1}
            maxLength={32}
            required
          />
          <button className={`${shared.primaryButton} ${shared.marginTopSmall}`}>Submit</button>
        </form>
      </div>
    </div>
  );
};

export { WinModal };
