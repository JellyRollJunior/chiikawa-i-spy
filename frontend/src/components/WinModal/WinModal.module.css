.modalWrapper {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;

  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(104, 104, 104, 0.5);
}

.modal {
  min-width: 250px;
  padding: 16px 12px;
  color: var(--dark-green);
  border-width: 5px;
}

.modal h2,
.modal p,
.modal label {
  color: inherit
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.form :first-child {
  align-self: center;
}

.form input {
  display: block;
  border-radius: 5px;
}