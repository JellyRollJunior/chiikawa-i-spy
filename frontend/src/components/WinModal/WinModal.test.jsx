import { vi, describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { WinModal } from './WinModal.jsx';
import { BrowserRouter } from 'react-router';

vi.mock('../IconWrapper/IconWrapper.jsx', () => ({
  IconWrapper: ({ children }) => {
    children;
  },
}));

describe('Win Modal component', () => {
  it('Renders win time, name input, and submit button', () => {
    const { container } = render(
      <BrowserRouter>
        <WinModal time={3} />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
