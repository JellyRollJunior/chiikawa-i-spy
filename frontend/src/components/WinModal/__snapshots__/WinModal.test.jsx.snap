// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Win Modal component > Renders win time, name input, and submit button 1`] = `
<div>
  <div
    class="_modalWrapper_e4f685"
  >
    <div
      class="_card_5890fc _modal_e4f685 _fadeIn_5890fc"
    >
      <form
        class="_form_e4f685"
      >
        <p>
          Wow! You found all the hidden critters in
           
          <strong
            class="_bold_5890fc"
          >
            3
             seconds
          </strong>
        </p>
        <p>
          Add your username to the leaderboards
        </p>
        <label
          class="_bold_5890fc _marginTopSmall_5890fc"
          for="name"
        >
          Name
        </label>
        <input
          id="name"
          maxlength="32"
          minlength="1"
          name="name"
          required=""
          type="text"
          value="Anonymous Mob Creature"
        />
        <button
          class="_primaryButton_5890fc _marginTopSmall_5890fc"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
